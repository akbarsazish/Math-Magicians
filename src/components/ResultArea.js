import PropTypes from 'prop-types';

const ResultArea = ({ result }) => <div type="text" className="resultArea">{result}</div>;
ResultArea.propTypes = {
  result: PropTypes.string.isRequired,
};

export default ResultArea;
