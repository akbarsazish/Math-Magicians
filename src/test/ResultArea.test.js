import * as React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import ResultArea from '../components/ResultArea';

describe('ResultArea', () => {
  it('should render the result area with the correct text', () => {
    render(<ResultArea result="This is the result" />);
    const resultArea = screen.getByTestId('resultArea');
    expect(resultArea).toBeInTheDocument();
    expect(resultArea.textContent).toBe('This is the result');
  });

  it('should throw an error if the result prop is not a string', () => {
    try {
      render(<ResultArea result="123" />);
    } catch (error) {
      expect(error.message).toBe(
        'The result prop must be a string. Received: 123',
      );
    }
  });
});
