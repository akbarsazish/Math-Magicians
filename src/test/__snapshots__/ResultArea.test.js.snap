// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultArea should render the result area with the correct text 1`] = `
<div
  class="resultArea"
  data-testid="resultArea"
  type="text"
>
  This is the result
</div>
`;
